src/Analyzers/CSharp/CodeFixes/TransposeRecordKeyword/CSharpTransposeRecordKeywordCodeFixProvider.cs

```diff
@@ -0,0 +1,135 @@
+                if (leadingTrivia.Count >= 2)
+                {
+                    var skippedTrivia = leadingTrivia[^2];
+                    whitespaceTrivia = leadingTrivia[^1];
+                    if (skippedTrivia.Kind() == SyntaxKind.SkippedTokensTrivia &&
+                        whitespaceTrivia.Kind() == SyntaxKind.WhitespaceTrivia)
+                    {
+                        var structure = (SkippedTokensTriviaSyntax)skippedTrivia.GetStructure()!;
+                        var tokens = structure.Tokens;
+                        if (tokens.Count == 1)
+                        {
+                            var lastSkippedToken = tokens.Single();
+                            if (lastSkippedToken.Kind() is SyntaxKind.ClassKeyword or SyntaxKind.StructKeyword)
+                            {
+                                classOrStructKeyword = lastSkippedToken;
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            classOrStructKeyword = default;
+            whitespaceTrivia = default;
+            return false;
+        }
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var diagnostic = context.Diagnostics.First();
+            if (TryGetRecordDeclaration(diagnostic, cancellationToken, out var recordDeclaration) &&
+                TryGetTokens(recordDeclaration, out _, out _, out _))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => this.FixAsync(document, diagnostic, c)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                if (TryGetRecordDeclaration(diagnostic, cancellationToken, out var recordDeclaration))
+                {
+                    editor.ReplaceNode(
+                        recordDeclaration,
+                        (current, _) =>
+                        {
+                            var currentRecordDeclaration = (RecordDeclarationSyntax)current;
+                            if (!TryGetTokens(currentRecordDeclaration, out var recordKeyword, out var classOrStructKeyword, out var whitespace))
+                                return currentRecordDeclaration;
+
+                            return currentRecordDeclaration
+                                .WithClassOrStructKeyword(classOrStructKeyword.WithAppendedTrailingTrivia(whitespace))
+                                .WithKeyword(recordKeyword.WithLeadingTrivia(
```

Comment:
Should trailing trivia on `recordKeyword` be transferred to `classOrStructKeyword`?

Reply:
yes.  that would atually make things substantially easier.  effectively we shuld keep all teh trivia fixed, and just swap the tokens (without trivia).  thanks!